requirements.txt

```diff
@@ -21,17 +21,17 @@ future==0.18.2
 hurry.filesize==0.9
 idna==2.10
 imagesize==1.2.0
-importlib-metadata==3.4.0
+importlib-metadata==3.7.0
 iniconfig==1.1.1
-ipython==7.20.0
+ipython==7.21.0
 ipython-genutils==0.2.0
 jedi==0.18.0
 Jinja2==2.11.3
 MarkupSafe==1.1.1
 mccabe==0.6.1
 more-itertools==8.7.0
 motor==2.3.1
-mypy==0.800
+mypy==0.812
```

Comment:
Since you're mypy'ing locally, can you do `pytest --mypy --flake8` for CirclecCI too?

Reply:
or something to that effect, since you're not using pytest

Reply:
So, CircleCI makes use of the `snake` script as part of its build process:
https://github.com/WIPACrepo/lta/blob/d9949713c2dfbda01d1a499ae7a7696c54d34e6b/.circleci/config.yml#L23

That `snake rebuild` maps to this `doRebuildTask` function:
https://github.com/WIPACrepo/lta/blob/d9949713c2dfbda01d1a499ae7a7696c54d34e6b/snake#L89

And that in turn calls this `doLintTask` function:
https://github.com/WIPACrepo/lta/blob/d9949713c2dfbda01d1a499ae7a7696c54d34e6b/snake#L79

Between the two of them, we're running pytest, flake8, and mypy.
We do skip mypy for the tests; I've got a note there `Maybe when pytest has a mypy library stub`

Is there something specific that should be changed for the CircleCI build?

Reply:
I'm unfamiliar with snake, but that sounds good!

Reply:
Snake is just a custom script @blinkdog wrote, to run multiple tests.  Nothing magical about it.

Reply:
> Snake is just a custom script @blinkdog wrote, to run multiple tests. Nothing magical about it.

Correct. It's intended as an analog to CoffeeScript's [cake](https://coffeescript.org/#cake) (which was in turn was an analog of Ruby's `rake`).

There's some pretty handy stuff in there:
```
(env) pmeade@bluebird ~/github/lta-dev $ snake
Try one of the following tasks:

snake check                # Check dependency package versions
snake circleci             # Perform CI build and test
snake clean                # Remove build cruft
snake coverage             # Perform coverage analysis
snake dist                 # Create a distribution tarball and wheel
snake docker               # Create a docker container
snake lint                 # Run static analysis tools
snake rebuild              # Test and lint the module
snake test                 # Test the module
```

If you look around in the source, please forgive the camel case. Those were my early days grappling with Python.
And even now, I see `Create a docker container` where I should have written `Build a Docker image`. :confounded: